RELEASE.md

```diff
@@ -358,3 +358,109 @@ actually push artifacts to the real release channels.
 Also note that there is no additional automation around this
 currently. If you want to initiate a fake release, you must take that
 action yourself.
+
+# "Yanking" a Release
+
+In general, if an escaped defect is discovered in a Habitat release,
+the first solution should be to "roll forward". That is, a patch that
+addresses the defect (either by fixing it or reverting the code
+responsible) should be merged to master, and then a new release is
+built from that. This is the desired solution because it requires no
+extraordinary measures by end-users to take advantage of the fix; they
+only need to update their software as they would for any "normal"
+release.
+
+In the rare case that that is _not_ possible (e.g., the cause of the
+defect is not readily apparent), you can remove or "yank" the
```

Comment:
I think it's important for us to enumerate the cases where a yank is the right call, so far I can think of:
1. The cause of the defect isn't apparent, but definitely isn't present in the previous release
1. The defect is severe enough that waiting for a new release isn't acceptable (becomes less an issue as our release automation matures)
1. A simple revert wouldn't work due to infrastructure dependencies such as schema changes and properly testing the forward fix will be too time-consuming

Reply:
Good point... I'll add this. These are basically all the reasons we had for the 0.80.0 release 😅